internal/backend/remote-state/http/helper.go

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) The OpenTofu Authors
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (c) 2023 HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package http
+
+import (
+	"encoding/json"
+	"io"
+	"log"
+	"net/http"
+)
+
+// Helper function to parse response body bytes into a string for logging
+func parseResponseBody(resp *http.Response) string {
```

Comment:
nit: `parseResponseBody` and `parseHeaders` should only be used for logging purposes and I would suggest reflecting this in their naming. Something like `prepareResponseBodyForLog` / `parseResponseBodyForLog`